orte/mca/schizo/ompi/schizo_ompi.c

```diff
@@ -908,9 +908,9 @@ static int setup_fork(orte_job_t *jdata,
 
     /* setup yield schedule - do not override any user-supplied directive! */
     if (oversubscribed) {
-        opal_setenv("OMPI_MCA_mpi_yield_when_idle", "1", false, &app->env);
+        opal_setenv("OMPI_MCA_mpi_oversubscribe", "1", true, &app->env);
```

Comment:
I know the previous code was directly setting an MPI env variable, but this feels wrong -- it's an abstraction violation for ORTE to set anything in the OMPI layer.

Reply:
I understand how that can feel wrong. That being said, doesn't an `ompi` component in `ORTE` feel wrong in the same way ? Since `orte` is scheduled to be replaced with `prrte`,  I did not want to use a MCA parameter related to `orte`. Would setting `OMPI_MCA_opal_oversubscribe` instead feel more right ?

Reply:
Let's not get too hung up on component names, guys - I could have called it anything. We even set "MPI_foo" envars in this component and freely note that it is an abstraction violation because the community long-ago requested that we do so.

Perhaps instead of defining an MCA parameter we could just pass something via PMIx in the job info to indicate oversubscription. Have to define a new attribute for it, but that's trivial to do, and it would translate easier to PRRTE. You'd need to see where you would want to access it and deal with setting the yield variable without overriding any user setting, but that should be doable.

Reply:
Actually, I should have looked closer and realized that this is a component exclusively for `ompi`.  I think that changes the equation here.  I retract my original comment.